server/server.go

```diff
@@ -414,6 +428,23 @@ func NewServer(opts *Options) (*Server, error) {
 	return s, nil
 }
 
+// clusterName returns our cluster name which could be dynamic.
+func (s *Server) ClusterName() string {
+	s.mu.Lock()
+	cn := s.info.Cluster
+	s.mu.Unlock()
+	return cn
+}
+
+// setClusterName will update the cluster name for this server.
+func (s *Server) setClusterName(name string) {
+	s.mu.Lock()
+	s.info.Cluster = name
```

Comment:
Since if this server accepts a remote's cluster name, I think that if the remote is not dynamic, then it should change its state to not dynamic so that when dealing with other servers it sends the info that the name is not dynamic.

Reply:
Agree.

Reply:
I think it is ok as is TBH. This is a temporary condition IMHO. Most clusters will have names set.